### ISSUE TITLE
`parseDateFields` skips date fields nested deeper than 3 levels

### ISSUE THREAD

**Issue by tobice** [apify/apify-client-js#522]
[`parseDateFields`](https://github.com/apify/apify-client-js/blob/980d958ee5e6bf8610f7ce08fb282e126c68fa82/src/utils.ts#L62) traverses the provided JSON (used for Apify API responses) and converts certain fields to a `Date` object. The traversion depth has been historically limited to 3, possibly to avoid cyclic dependencies and improve performance on large responses.

This limitation is not communicated in any way, which might lead to weird surprises. We tried adding a `log.warning` but that led to log pollution as there are objects with arbitrary structure that went over the depth limit (most likely user data, data sets). Context [here](https://apifier.slack.com/archives/CGZSN9DQC/p1708707905914559).

Possible solutions:
- Remove the limitation
- Stop parsing objects with arbitrary structure (user data, data sets)

Both are potential breaking changes.

This is currently not causing any issues and can be treated with low priority.

**Comment by drobnikj** [comment]
Let's check with @B4nan, but I guess this is low-prio for now.

**Comment by B4nan** [comment]
Yeah, this can wait, we just wanted to have this tracked. It sounds like a good first issue for some new joiners.

**Comment by vdusek** [comment]
resolved by https://github.com/apify/apify-client-js/pull/1060
